src/main/java/org/jfrog/bamboo/task/ArtifactoryNpmTask.java

```diff
@@ -56,15 +53,19 @@ public ArtifactoryNpmTask(EnvironmentVariableAccessor environmentVariableAccesso
     }
 
     @Override
-    protected void initTask(@NotNull TaskContext context) throws TaskException {
-        taskContext = context;
-        logger = taskContext.getBuildLogger();
+    protected void initTask(@NotNull CommonTaskContext context) {
+        super.initTask(context);
         npmBuildContext = new NpmBuildContext(taskContext.getConfigurationMap());
         buildParamsOverrideManager = new BuildParamsOverrideManager(customVariableContext);
-        buildName = npmBuildContext.getBuildName(taskContext.getBuildContext());
-        buildNumber = npmBuildContext.getBuildNumber(taskContext.getBuildContext());
-        initBuildInfoHelper();
-        environmentVariables = getEnv();
+        BuildContext buildContext = ((TaskContext) context).getBuildContext();
+        buildName = npmBuildContext.getBuildName(buildContext);
+        buildNumber = npmBuildContext.getBuildNumber(buildContext);
+        initBuildInfoHelper(buildContext);
+        try {
+            environmentVariables = getEnv();
```

Comment:
What is the reason for us wanting to skip this exception and catch it?

Reply:
You're right. We should not catch this exception.